lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -478,10 +477,10 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
         final MethodCall decoded = codec.decodeMethodCall(data);
         switch (decoded.method) {
           case 'SystemNavigator.pop':
-            // TODO(gspencergoog): As multi-window support expands, the pop call
+            // TODO(a-wallen): As multi-window support expands, the pop call
             // will need to include the window ID. Right now only one window is
             // supported.
-            (_windows[0]! as EngineFlutterWindow)
+            (viewData[0]! as EngineFlutterWindow)
```

Comment:
`0` seems to have a special meaning here that's not documented.

The code looks like it's trying to access the "first" element in a list, but that's not what's happening. `viewData` (and previously `_windows`) is a map, and the key `0` is basically a special hard-coded key that's shared between here and:
https://github.com/flutter/engine/blob/bb4015269d3d252160133b377b72e315e89f2d4d/lib/web_ui/lib/src/engine/window.dart#L366-L367

I suggest we make that explicit through a constant (e.g. `const int kSingletonViewId = 0`), and stop using literal `0` elsewhere.

(I know this existed before your PR, but I wanted to recommend it since you are working on this part of the code anyway).

Reply:
https://github.com/flutter/engine/pull/38453/commits/8fa153bb72ad174e862c2f92ef886a74788213e3